kinesis/kinesis.go

```diff
@@ -637,6 +646,79 @@ func gzipCompress(data []byte) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
+func compressThenTruncate(compressorFunc CompressorFunc, data []byte, maxOutLen int, truncatedSuffix []byte, outputPlugin OutputPlugin) ([]byte, error) {
```

Comment:
could you add a comment about what this func is doing.

Reply:
Sure. Good idea.